xarray/plot/utils.py

```diff
@@ -165,6 +165,10 @@ def _determine_cmap_params(plot_data, vmin=None, vmax=None, cmap=None,
 
     calc_data = np.ravel(plot_data[~pd.isnull(plot_data)])
 
+    # Handle all-NaN input data gracefully
+    if calc_data.size == 0:
+        calc_data = np.array(0.0)
```

Comment:
I suppose the colorbar get the wrong data in this case, but it's still better than an error message?

Again, please comment that the value 0 is arbitrary.

Reply:
There *is* no data to give the colorbar in this case, so it uses Matplotlib defaults.  Works beautifully for faceted plots of image timeseries though 😄